Source/WebCore/dom/SerializedNode.h

```diff
@@ -25,35 +25,54 @@
 
 #pragma once
 
+#include "SecurityOriginData.h"
 #include <wtf/text/WTFString.h>
 
+namespace JSC {
+class JSGlobalObject;
+class JSValue;
+}
+
 namespace WebCore {
 
+class Document;
+class JSDOMGlobalObject;
+
+enum class ClonedDocumentType : uint8_t;
+
 struct SerializedNode {
     struct Attr {
         String prefix;
         String localName;
         String namespaceURI;
         String value;
     };
-    struct Document {
-        // FIXME: Implement.
+    struct ContainerNode {
+        Vector<SerializedNode> children;
+    };
+    struct Document : public ContainerNode {
+        ClonedDocumentType type;
+        URL url;
+        URL baseURL;
+        URL baseURLOverride;
+        Variant<String, URL> documentURI;
+        String contentType;
```

Comment:
FWIW, the specification also lists encoding and allow declarative shadow roots as aspects to be copied. (And origin and custom element registry, but I don't think those are applicable here.)

Reply:
I don't think we've implemented "allow declarative shadow roots" yet.  I arbitrarily decided to inherit the encoding from the deserialization document, but may revisit.